contrib/babelfishpg_tsql/src/fts_parser.y

```diff
@@ -195,6 +205,93 @@ static char
     return output.data;
 }
 
+/* Helper function that takes in a prefix word or phrase and returns the same prefix word/phrase in Postgres format
+ * Example: '"word*"' is rewritten into 'word:*'; '"word1 word2 word3*"' is rewritten into 'word1:*<->word2:*<->word3:*'
+ * Case 1: '"word*"' = 'word:*'
+ * Case 2: '"word1 word2 word3*"' = 'word1:*<->word2:*<->word3:*'
+ * Case 3: '  "word*"' = 'word:*' || '"word*" ' = 'word:*' || ' "word*" ' = 'word:*'
+ * Case 4: '" word1 word2*"' = 'word1:*<->word2:*'
+ * Case 5: '"word1* word2*"' = 'word1:*<->word2:*'
+ * Case 6: '"word1 word2* "' && '" word1 word2* "' are treated as simple terms over SQL server
+ * Trivial Case: spaces before and after double quotes, Example - '   "word1 word2" ' = 'word1<->word2'
+ */
+static char
+*translate_prefix_term(const char* inputStr) {
+    char                *outputStr;
+    char                *trimmedOutputStr;
+    StringInfoData      output;
+    static const char   *specialChars = "~!&|@#$%^+=\\;:<>?.\\/`'_";
+
+
+    /* Check for empty input - this should not be possible based on lexer rules, but check just in case */
+    if (!inputStr || !*inputStr) {
+      ereport(ERROR,
+          (errcode(ERRCODE_INTERNAL_ERROR),
+           errmsg("Null or empty full-text predicate.")));
+    }
+    
+    outputStr = pstrdup(inputStr);
+
+    /* removing leading and trailing spaces outside of double quotes */
+    trim(outputStr, false);
+  
+    /* 
+     * removing leading spaces, for the phrase enclosed in double quotes
+     * search string with trailing spaces are identified as simple terms by the lexer
+     */
+    trim(outputStr, true);
+    
+    initStringInfo(&output);
+    trimmedOutputStr = outputStr;
+
+    /* 
+     * Rewriting search string in format word1:*<->word2:*  
+     */
+    while(*trimmedOutputStr) {
+        if (strchr(specialChars, *trimmedOutputStr) != NULL) {
+            ereport(ERROR,
+                (errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+                errmsg("Special characters in the prefix term search condition are not currently supported in Babelfish")));
+        }
```

Comment:
When an error is thrown here, both `outputStr` and `output.data` memory will be leaked. Please free these allocations before calling ereport().